bootstrap/src/sun/nio/ch/lincheck/EventTracker.kt

```diff
@@ -44,7 +43,7 @@ interface EventTracker {
 
     fun afterReflectiveSetter(receiver: Any?, value: Any?)
 
-    fun beforeMethodCall(owner: Any?, className: String, methodName: String, codeLocation: Int, params: Array<Any?>)
+    fun beforeMethodCall(owner: Any?, className: String, methodName: String, codeLocation: Int, methodId: Int, params: Array<Any?>)
```

Comment:
No, it shouldn't. The only reason for methodId is to understand in case of the recursive spin lock the depth of the spin cycle start. We compare them when a walking up by the stack of the trace points. But in case of the atomicMethods no trace point can exist inside this method, so methodId is useless. That's why we don't pass it in beforeAtomicMethodCall.

Reply:
Ok, I understood that you don't need the method ids of atomic methods for the spin-lock detection algorithm. 
But from the API point of view it looks a bit strange:

```kotlin
fun beforeMethodCall(owner: Any?, className: String, methodName: String, codeLocation: Int, methodId: Int, params: Array<Any?>)

fun beforeAtomicMethodCall(owner: Any?, className: String, methodName: String, codeLocation: Int, params: Array<Any?>)
``` 

We now have two similar methods, which have almost exactly the same signature, except that `beforeMethodCall` takes additional argument of `methodId`, but `beforeAtomicMethodCall` does not. 

Atomic methods can be seen as a sub-kind of all methods, so it is natural to expect that they tracker methods provide to the strategy all the information that tracker for other methods provide.

Reply:
Do we need `beforeAtomicMethodCall` at all?

Reply:
As was discussed with `@ndkoval`, we are going to remove `beforeAtomicMethodCall`, leave only `beforeMethodCall`, and detect atomic primitives dynamically. 

Another reason why we want to do that is because currently we do not handle classes inheriting from `Atomic*` classes correctly, because their methods are not detected as "atomic" statically by the bytecode transformer. Thus we want to perform the type-check dynamically. 

But all of these problems are orthogonal to this PR, I am currently working on a solution in a separate PR.

Reply:
@eupp should we wait for your solution or merge this PR first?

Reply:
I do not see any reason to wait, these two PRs are orthogonal.

Reply:
We have removed `beforeAtomicMethodCall` first.